SPDY/SPDYProtocol.h

```diff
@@ -288,3 +293,22 @@ extern NSString *const SPDYMetadataStreamTxBytesKey;
 @property NSInteger proxyPort;
 
 @end
+
+/**
+  Protocol that may be implemented by the delegate property of NSURLSession.
+ 
+  This will provide additional context for the request, if desired. Implementing
+  it is optional, and only applies for NSURLSession-based requests. All calls 
+  made using the NSOperationQueue set in delegateQueue in NSURLSession, or else 
+  the mainQueue if one is not set.
+*/
+@protocol SPDYURLSessionDelegate <NSObject>
+
+/**
+  Called just before the request is dispatched and provides the SPDYProtocol 
+  instance handling the request.
+*/
+@optional
+- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didStartLoadingRequest:(NSURLRequest *)request withSPDYProtocol:(SPDYProtocol *)protocol;
```

Comment:
Honestly, I've been considering this too.  I think we'd be better providing a separate object via a protocol.

```
- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didStartLoadingRequest:(NSURLRequest *)request withSPDYProtocolContext:(id<SPDYProtocolContext>)context;
```